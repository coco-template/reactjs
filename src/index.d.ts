// internal
import { rootReducer } from './redux';

declare module '*.pcss';

// global, avoid redundant unnecessary import
declare global {
  declare type AppState = ReturnType<typeof rootReducer>;
}
