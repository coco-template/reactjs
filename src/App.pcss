/* 样式覆盖 */
:global(.main) {
  height: 100%;
}

/* 常规布局 */
.package {
  min-height: 100%;
}

.sider {
  position: fixed;
  min-height: 100%;
  overflow: auto;
}

.content {
  margin-left: 200px;
}

.link {
  display: inline-block !important;
}

.menu-item {
  margin-top: 0 !important;
  margin-bottom: 0 !important;
}
