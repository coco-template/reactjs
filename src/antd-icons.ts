/**
 * @description - cherry pick antd icons
 * @author - huang.jian <hjj491229492@hotmail.com>
 */

export { default as UserOutlined } from '@ant-design/icons/lib/icons/UserOutlined';
