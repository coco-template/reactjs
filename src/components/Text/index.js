/**
 * @description - nothing but export component
 */
export { default } from './Text';
