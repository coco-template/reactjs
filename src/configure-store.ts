/**
 * @description - redux connection
 * @author - huang.jian <hjj491229492@hotmail.com>
 */

// package
import { applyMiddleware, compose, createStore, Middleware } from 'redux';
import { createEpicMiddleware } from 'redux-observable';
// internal
import { rootEpic, rootReducer } from './redux';
// state
export type AppState = ReturnType<typeof rootReducer>;

export function configureStore() {
  const middleware = {
    epic: createEpicMiddleware(),
  };
  const middleWares: Middleware[] = [middleware.epic];
  const middleWaresEnhancer = applyMiddleware(...middleWares);
  // redux-devtools-extension 引入
  const devtoolsExtensionCompose =
    typeof __REDUX_DEVTOOLS_EXTENSION_COMPOSE__ === 'function'
      ? __REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
          features: {
            persist: false,
            dispatch: true,
          },
        })
      : compose;
  const composeEnhancers =
    process.env.NODE_ENV === 'production' ? compose : devtoolsExtensionCompose;

  const store = createStore(rootReducer, composeEnhancers(middleWaresEnhancer));

  // redux-observable setup
  middleware.epic.run(rootEpic);

  if (module.hot) {
    module.hot.accept('./redux', () => store.replaceReducer(rootReducer));
  }

  return store;
}

export const rootStore = configureStore();
