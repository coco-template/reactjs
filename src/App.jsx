/**
 * @description - develop web page component
 * @author - huang.jian <hjj491229492@hotmail.com>
 */
// External
import React, { Fragment } from 'react';
import {
  BrowserRouter,
  NavLink,
  Redirect,
  Route,
  Switch,
} from 'react-router-dom';
import { Layout, Menu, Icon } from 'antd';
import { hot } from 'react-hot-loader';
// Internal
import { Text } from './components';
import styles from './App.pcss';
// scope
const { Sider, Content } = Layout;
const showcases = {
  Text: () => (
    <Fragment>
      <Text theme="primary">Lift is short, enjoy yourself!</Text>
      <Text theme="info">Lift is short, enjoy yourself!</Text>
      <Text theme="success">Lift is short, enjoy yourself!</Text>
      <Text theme="warning">Lift is short, enjoy yourself!</Text>
      <Text theme="danger">Lift is short, enjoy yourself!</Text>
    </Fragment>
  ),
};

function App() {
  return (
    <BrowserRouter>
      <Layout className={styles.package}>
        <Sider className={styles.sider}>
          <Menu
            mode="inline"
            theme="dark"
            defaultSelectedKeys={['package.text']}
          >
            <Menu.Item key="package.text" className={styles.menuItem}>
              <Icon type="user" />
              <NavLink to="/package/text" className={styles.link}>
                Text
              </NavLink>
            </Menu.Item>
          </Menu>
        </Sider>
        <Content className={styles.content}>
          <Switch>
            <Route path="/package/text" component={showcases.Text} />
            <Redirect to="/package/text" />
          </Switch>
        </Content>
      </Layout>
    </BrowserRouter>
  );
}

export default hot(module)(App);
