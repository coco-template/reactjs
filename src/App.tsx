/**
 * @description - develop web page component
 * @author - huang.jian <hjj491229492@hotmail.com>
 */

// external
import React, { lazy } from 'react';
import { hot } from 'react-hot-loader';
import { BrowserRouter, Redirect, Route, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';

// internal
import { rootStore } from './configure-store';
import withLazySuspense from './HOC/withLazySuspense';
import './App.pcss';
// scope
const LazyHistoryWrapper = lazy(() =>
  import(/* webpackChunkName: 'history' */ './pages/History')
);
const LazyDeviceWrapper = lazy(() =>
  import(/* webpackChunkName: 'history' */ './pages/Device')
);
const LazyHistory = withLazySuspense(LazyHistoryWrapper);
const LazyDevice = withLazySuspense(LazyDeviceWrapper);

function App() {
  return (
    <Provider store={rootStore}>
      <BrowserRouter>
        <Switch>
          <Route path="/pwa/history" component={LazyHistory} />
          <Route path="/pwa/device" component={LazyDevice} />
          <Redirect to="/pwa/history" />
        </Switch>
      </BrowserRouter>
    </Provider>
  );
}

export default hot(module)(App);
