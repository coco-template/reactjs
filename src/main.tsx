/**
 * @description - just render real time component
 * @author - huang.jian <hjj491229492@hotmail.com>
 */

// external
import React from 'react';
import { render } from 'react-dom';

// internal
import App from './App';

// scope
const container = document.querySelector('.main');

render(<App />, container);

// register service worker only in production
if (process.env.NODE_ENV === 'production') {
  if ('serviceWorker' in navigator) {
    window.addEventListener('load', () => {
      navigator.serviceWorker
        .register('/service-worker.js')
        .then(() => {
          // tslint:disable-next-line:no-console
          console.log('service worker register success!');
        })
        .catch(() => {
          // tslint:disable-next-line:no-console
          console.log('service worker register failure!');
        });
    });
  }
}
