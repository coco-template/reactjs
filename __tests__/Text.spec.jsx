// external
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// internal
import { Text } from '../src/components';

describe('Text Component', () => {
  beforeAll(() => {
    configure({ adapter: new Adapter() });
  });

  it('should render standard primary text', () => {
    const wrapper = shallow(<Text theme="primary">Jest Unit Test</Text>);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should accept align props', () => {
    const wrapper = shallow(
      <Text theme="primary" align="center">
        Jest Unit Test
      </Text>
    );

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should accept transform props', () => {
    const wrapper = shallow(
      <Text theme="primary" transform="uppercase">
        Jest Unit Test
      </Text>
    );

    expect(wrapper.html()).toMatchSnapshot();
  });
});
