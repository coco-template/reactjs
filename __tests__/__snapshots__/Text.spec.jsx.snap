// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component should accept align props 1`] = `"<p class=\\"base primary center noop \\">Jest Unit Test</p>"`;

exports[`Text Component should accept transform props 1`] = `"<p class=\\"base primary left uppercase \\">Jest Unit Test</p>"`;

exports[`Text Component should render standard primary text 1`] = `"<p class=\\"base primary left noop \\">Jest Unit Test</p>"`;
